cmd/kubeadm/app/cmd/util/join_test.go

```diff
@@ -0,0 +1,240 @@
+			token:       "test-token",
+			expectError: false,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			tmpDir, err := os.MkdirTemp("", "kubeadm-images-test")
+			if err != nil {
+				t.Fatalf("Unable to create temporary directory: %v", err)
+			}
+			defer func() {
+				err := os.RemoveAll(tmpDir)
+				if err != nil {
+					t.Fatalf("Unable to remove temporary directory: %v", err)
+				}
+			}()
+
+			configFilePath := filepath.Join(tmpDir, "test-config-file")
+			err = writeKubeConfigToFile(tt.kubeConfig, configFilePath)
+			require.NoError(t, err)
+
+			_, err = GetJoinWorkerCommand(configFilePath, tt.token, false)
+
+			if tt.expectError {
+				require.Error(t, err)
+				require.Contains(t, err.Error(), tt.errorMessage)
+			} else {
+				require.NoError(t, err)
+			}
+		})
+	}
+}
+
+func TestGetJoinControlPlaneCommand(t *testing.T) {
+	tests := []struct {
+		name         string
+		kubeConfig   *clientcmdapi.Config
+		expectError  bool
+		errorMessage string
+		token        string
+	}{
+		{
+			name:        "Success with valid kubeconfig and token",
+			kubeConfig:  generateValidKubeConfig(),
+			token:       "test-token",
+			expectError: false,
+		},
+		{
+
+			name:         "Error to load kubeconfig",
+			kubeConfig:   nil,
+			token:        "test-token",
+			expectError:  true,
+			errorMessage: "failed to load kubeconfig",
+		},
+
+		{
+			name:         "Error to get default cluster config",
+			kubeConfig:   &clientcmdapi.Config{},
+			token:        "test-token",
+			expectError:  true,
+			errorMessage: "failed to get default cluster config",
+		},
+		{
+			name:         "Error when CA certificate is invalid",
+			kubeConfig:   generateInvalidCAKubeConfig(),
+			expectError:  true,
+			errorMessage: "failed to parse CA certificate from kubeconfig",
+		},
+		{
+			name:         "Error when CA certificate file path is invalid",
+			kubeConfig:   generateInvalidFilePathKubeConfig(),
+			token:        "test-token",
+			expectError:  true,
+			errorMessage: "failed to load CA certificate referenced by kubeconfig",
+		},
+		{
+			name:         "Error when CA certificate is missing",
+			kubeConfig:   generateMissingCAKubeConfig(),
+			token:        "test-token",
+			expectError:  true,
+			errorMessage: "no CA certificates found in kubeconfig",
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			tmpDir, err := os.MkdirTemp("", "kubeadm-images-test")
+			if err != nil {
+				t.Fatalf("Unable to create temporary directory: %v", err)
+			}
+			defer func() {
+				err := os.RemoveAll(tmpDir)
+				if err != nil {
+					t.Fatalf("Unable to remove temporary directory: %v", err)
+				}
+			}()
+
+			configFilePath := filepath.Join(tmpDir, "test-config-file")
+			if tt.kubeConfig != nil {
+				err = writeKubeConfigToFile(tt.kubeConfig, configFilePath)
+				require.NoError(t, err)
+			}
+
+			_, err = GetJoinControlPlaneCommand(configFilePath, tt.token, "", true, true)
```

Comment:
should we verify the output string for err != nil?

Reply:
Down